docs/user/managing-devices.md

```diff
@@ -0,0 +1,586 @@
+
+THese providers are described in the following.
+
+### Getting Configuration from a Git Repository
+
+You can store device configuration in a Git repository such as GitHub or GitLab and let Flight Control synchronize it to the device's file system by adding a Git Config Provider.
+
+The Git Config Provider takes the following parameters:
+
+| Parameter | Description |
+| --------- | ----------- |
+| Repository | The name of a Repository resource defined in Flight Control. |
+| TargetRevision | The branch, tag, or commit of the repository to checkout. |
+| Path | The subdirectory of the repository that contains the configuration. |
+| MountPath | (Optional) The directory in the device's file system to write the configuration to. Defaults to the file system root `/`. |
+
+The Repository resource definition tells Flight Control the Git repository to connect to and which protocol and access credentials to use. It needs to be set up once (see Setting Up Repositories) and can then be used to configure multiple devices or fleets.
+
+#### Example
+
+A assume a Git repository `github.com/flightctl/flightctl-demos` that in its branch `main` stores device network and time server configuration organized by deployment site as follows:
+
+```console
+.
+├── factory-a
+│   └── etc
+│       ├── chrony.conf
+│       └── NetworkManager
+│           └── system-connections
+│               └── wifi-access.nmconnection
+└── factory-b
+    └── etc
+        ├── chrony.conf
+        └── NetworkManager
+            └── system-connections
+                └── wifi-access.nmconnection
+```
+
+First, create a new Repository resource. Give it a name that you can later reference in your configurations, for example "site-settings", and select the Repository type "git". Next, enter the URL of your git repository, in this case `https://github.com/flightctl/flightctl-demos.git`.
+
+You can now reference this Repository when you configure devices. For example, to apply the configuration files under `/factory-a` to a device, add a Git Config Provider to the device's specification using the following parameters:
+
+| Parameter | Value |
+| --------- | ----- |
+| Repository | site-settings |
+| TargetRevision | main |
+| Path | /factory-a |
+| MountPath | (leave empty to use the default `/` root directory) |
+
+### Getting Secrets from a Kubernetes Cluster
+
+You can let Flight Control query a Kubernetes cluster for a Kubernetes Secret. The content of that Secret can then be written to a path on the device file system.
+
+The Kubernetes Secret Provider takes the following parameters:
+
+| Parameter | Description |
+| --------- | ----------- |
+| Name | The name of the Secret. |
+| NameSpace | The namespace of the Secret. |
+| MountPath | (Optional) The directory in the device's file system to write the secret's content to. Defaults to the file system root `/`. |
```

Comment:
We should consider making MountPath required because / is read-only.

Reply:
Good catch! In Secrets the MountPath *is* actually already required. Will fix.

I was going for consistency with the Inline and Git config, but for these using (or defaulting to) absolute paths is deliberate.

What is the behaviour when the mountPath contains a read-only location?